### PR TITLE
BUG: Fix indexing on DatetimeBlock

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Addresses one of the problems behind #26864, but probably doesn't address the statefullness problem there.

### CODE REVIEW

File: pandas/core/internals/blocks.py [hunk 1]
```diff
@@ -1539,6 +1539,10 @@ def iget(self, col):
             col, loc = col
             if not com.is_null_slice(col) and col != 0:
                 raise IndexError("{0} only contains one item".format(self))
+            if isinstance(col, slice):
```
Comment (jreback) [inline pandas/core/internals/blocks.py]: could be elif

File: pandas/tests/groupby/aggregate/test_other.py [hunk 2]
```diff
@@ -421,11 +421,12 @@ def test_agg_timezone_round_trip():
     assert ts == grouped.nth(0)['B'].iloc[0]
     assert ts == grouped.head(1)['B'].iloc[0]
     assert ts == grouped.first()['B'].iloc[0]
-    assert ts == grouped.apply(lambda x: x.iloc[0])[0]
+    assert ts == grouped.apply(lambda x: x.iloc[0]).iloc[0, 0]
```
Comment (jreback) [inline pandas/tests/groupby/aggregate/test_other.py]: can you reference this issue and maybe put a comment here on whey this is like this

Comment (jreback) [reply pandas/tests/groupby/aggregate/test_other.py]: or this is actually returning a single column DataFrame right? can you assert that

File: pandas/tests/groupby/aggregate/test_other.py [hunk 3]
```diff
@@ -421,11 +421,12 @@ def test_agg_timezone_round_trip():
...
 
     ts = df['B'].iloc[2]
     assert ts == grouped.last()['B'].iloc[0]
-    assert ts == grouped.apply(lambda x: x.iloc[-1])[0]
+    # FIXME: the next assertion is wrong; GH#26864
```
Comment (jreback) [inline pandas/tests/groupby/aggregate/test_other.py]: same here

Comment (jreback) [reply pandas/tests/groupby/aggregate/test_other.py]: also ok with moving these to a new test if easier

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jbrockmendel** [comment]
I'm pretty sure this problem also affects other EA blocks, will check and add tests if so

**Comment by jbrockmendel** [comment]
@jreback any idea what’s up with the CI?

**Comment by jreback** [comment]
```
/home/vsts/work/1/s/doc/source/whatsnew/v0.25.0.rst:786: WARNING: Unknown interpreted text role "method".
/home/vsts/work/1/s/doc/source/whatsnew/v0.25.0.rst:786: WARNING: Unknown interpreted text role "method".
```

**Comment by jreback** [comment]
method -> meth in the whatnew

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jbrockmendel

**Pull request merged**